projects/pswg/src/main/java/com/parzivail/pswg/client/particle/ExplosionSmokeParticle.java

```diff
@@ -0,0 +1,77 @@
+package com.parzivail.pswg.client.particle;
+
+import com.parzivail.util.client.particle.DecalParticle;
+import com.parzivail.util.client.particle.PParticleType;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.client.particle.*;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.util.math.MathHelper;
+import org.jetbrains.annotations.Nullable;
+
+@Environment(value= EnvType.CLIENT)
+public class ExplosionSmokeParticle extends SpriteBillboardParticle
+{
+	private static final int NUM_VARIANTS = 12;
+	private final int variant;
+	protected ExplosionSmokeParticle(ClientWorld clientWorld, double x, double y, double z, double vX, double vY, double vZ, SpriteProvider spriteProvider)
```

Comment:
Newline here, in general make sure you're following the code style of the rest of the repo